include/soci/soci-backend.h

```diff
@@ -234,6 +234,28 @@ class session_backend
         return false;
     }
 
+    // There is a set of standard SQL metadata structures that can be
+    // queried in a portable way - backends that are standard compliant
+    // do not need to override the following methods, which are intended
+    // to return a proper query for basic metadata statements.
+    virtual std::string get_table_names_query() const
+    {
+        return "select table_name as \"TABLE_NAME\""
+            " from information_schema.tables"
+            " where table_schema = 'public'";
+    }
+    virtual std::string get_column_descriptions_query() const
```

Comment:
Should have a comment documenting that this query takes a parameter (`t`) as it's easy to miss it otherwise.